lib/pg_eventstore/web/application.rb

```diff
@@ -0,0 +1,121 @@
+# frozen_string_literal: true
+
+require 'securerandom'
+
+module PgEventstore
+  module Web
+    class Application < Sinatra::Base
+      set :static_cache_control, [:private, max_age: 86400]
+      set :environment, -> { (ENV['RACK_ENV'] || ENV['RAILS_ENV'] || ENV['APP_ENV'])&.to_sym || :development }
+      set :logging, -> { environment == :development || environment == :test }
+      set :erb, layout: :'layouts/application'
+      set :sessions, true
+      set :session_secret, ENV.fetch('SECRET_KEY_BASE') { SecureRandom.hex(64) }
+
+      helpers(Paginator::Helpers) do
+        # @return [Array<Hash>, nil]
+        def streams_filter
+          params in { filter: { streams: Array => streams } }
+          streams&.select { _1 in { context: String, stream_name: String, stream_id: String } }&.map do
+            Hash[_1.reject { |_, value| value == '' }].transform_keys(&:to_sym)
+          end&.reject { _1.empty? }
+        end
+
+        # @return [Array<String>, nil]
+        def events_filter
+          params in { filter: { events: Array => events } }
+          events&.select { _1.is_a?(String) && _1 != '' }
+        end
+
+        # @return [Symbol]
+        def current_config
+          PgEventstore.available_configs.include?(session[:current_config]) ? session[:current_config] : :default
+        end
+      end
+
+      get '/' do
+        @collection = Paginator::EventsCollection.new(
+          current_config,
+          starting_id: params[:starting_id]&.to_i,
+          per_page: Paginator::EventsCollection::PER_PAGE[params[:per_page]],
+          order: Paginator::EventsCollection::SQL_DIRECTIONS[params[:order]],
+          options: { filter: { event_types: events_filter, streams: streams_filter } }
+        )
+        erb :'home/dashboard'
+      end
+
+      post '/change_config' do
+        config = params[:config]&.to_sym
+        config = :default unless PgEventstore.available_configs.include?(config)
+        session[:current_config] = config
+        redirect(url('/'))
+      end
+
+      get '/stream_contexts_filtering' do
+        content_type 'application/json; charset="utf-8"'
```

Comment:
I don't know sinatra much but maybe you can use https://github.com/sinatra/sinatra?tab=readme-ov-file#conditions

```
set(:json_content_type) { content_type 'application/json; charset="utf-8"' }

get '/stream_contexts_filtering', json_content_type: true do
  ...
end
```